src/Compilers/Core/MSBuildTask/Microsoft.CSharp.Core.targets

```diff
@@ -3,10 +3,22 @@
 <Project ToolsVersion="14.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <Import Project="Microsoft.Managed.Core.targets"/>
 
-  <PropertyGroup Condition="('$(TargetFrameworkIdentifier)' != '.NETCoreApp' OR '$(_TargetFrameworkVersionWithoutV)' &lt; '3.0') AND
-                            ('$(TargetFrameworkIdentifier)' != '.NETStandard' OR '$(_TargetFrameworkVersionWithoutV)' &lt; '2.1')">
-    <MaxSupportedLangVersion Condition="'$(MaxSupportedLangVersion)' == ''">7.3</MaxSupportedLangVersion>
-    <LangVersion Condition="'$(LangVersion)' == ''">$(MaxSupportedLangVersion)</LangVersion>
+  <PropertyGroup>
+    <!-- .NETCoreApp < 3.0, .NETStandard < 2.1, or any other target framework -->
+    <MaxSupportedLangVersion Condition="('$(TargetFrameworkIdentifier)' != '.NETCoreApp' OR '$(_TargetFrameworkVersionWithoutV)' &lt; '3.0') AND
+                                        ('$(TargetFrameworkIdentifier)' != '.NETStandard' OR '$(_TargetFrameworkVersionWithoutV)' &lt; '2.1') AND
+                                        '$(MaxSupportedLangVersion)' == ''">7.3</MaxSupportedLangVersion>
+    
+    <!-- .NETCoreApp < 5.0, .NETStandard == 2.1 -->
+    <MaxSupportedLangVersion Condition="('$(TargetFrameworkIdentifier)' != '.NETCoreApp' OR '$(_TargetFrameworkVersionWithoutV)' &lt; '5.0') AND
+                                        ('$(TargetFrameworkIdentifier)' != '.NETStandard' OR '$(_TargetFrameworkVersionWithoutV)' == '2.1') AND
+                                        '$(MaxSupportedLangVersion)' == ''">8.0</MaxSupportedLangVersion>
+
+    <!-- .NETCoreApp == 5.0 -->
+    <MaxSupportedLangVersion Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp' AND '$(_TargetFrameworkVersionWithoutV)' == '5.0' AND
```

Comment:
Have u verified that these are the values that will be passed for `net5.0`? Also what happens when we have the OS specific versions of the TF moniker like `net5.0-ios13`?

Reply:
Yes. `net5.0` will still have a target framework of `.NETCoreApp` [by design](https://github.com/dotnet/designs/blob/master/accepted/2020/net5/net5.md#why-is-net-50s-tfi-still-mapped-to-netcoreapp). the `-ios13` stuff isn't happening in .net 5 timeline now, it got pushed back to net6. Ultimately though, that information will be mapped to [different properties](https://github.com/dotnet/designs/blob/master/accepted/2020/net5/net5.md#mapping-to-properties)

---
In reply to: [440425405](https://github.com/dotnet/roslyn/pull/44911#discussion_r440425405) [](ancestors = 440425405)